### DISCUSSION TITLE
Missing user data in session from Auth0

### DISCUSSION THREAD

**Discussion by softshipper** [nextauthjs/next-auth#1834] [Help] [answered]
Hi all 

I am trying to use https://auth0.com/ as the provider and it does not work as expected.  I took the example application from https://github.com/nextauthjs/next-auth-example and allow only Auth0 to sign in.
Here is the source code:
[next-auth.zip](https://github.com/nextauthjs/next-auth/files/6367905/next-auth.zip)

After successfully signed in, NextAuth does not get session data from Auth0. I have recorded as video:
https://www.loom.com/share/ab4bf457075345feb78f375d0094e43b

As you can see, after the login it does not show the correct information. It shows **You are not signed in**, which  
`const [ session, loading ] = useSession()` does not get any data.

I have got also weird URL http://localhost:3000/?error=redirect_uri_mismatch&error_description=The+redirect_uri+MUST+match+the+registered+callback+URL+for+this+application.&error_uri=https%3A%2F%2Fdocs.github.com%2Fapps%2Fmanaging-oauth-apps%2Ftroubleshooting-authorization-request-errors%2F%23redirect-uri-mismatch&state=A1swBji56S8ZYdl5NAilfuMXldTNtJQ0

The settings of the identity provider in Auth0:
![Screenshot from 2021-04-23 22-36-49](https://user-images.githubusercontent.com/56756949/115928509-7f1b4a80-a486-11eb-8b74-6f2cefc4b1ef.png)

I have also tried on https://next-auth-example.vercel.app/ Auth0 as provider and it works fine.   

What am I doing wrong?  

Thanks

**Comment by balazsorban44** [comment, marked as answer]
check `jwt` and `session` https://next-auth.js.org/configuration/callbacks

Also, you say you use auth0, so you must configure your auth0 application against next-auth and github against auth0. not github against next-auth

**Reply by softshipper** [reply]
This was my mistake.
`github against auth0.` 

Thanks a lot
